File: problems/034-first-last-position-in-sorted-array/main.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        result = [0, 0]
        result[0] = self.find_left_index(nums, target)
        result[1] = self.find_right_index(nums, target)
        return result
    
    def find_left_index(self, arr, target):
        index = -1
        low = 0
        high = len(arr) - 1
        while (low <= high):
            mid = int(low + (high - low) / 2)
            if (arr[mid] >= target):
                high = mid - 1
            else:
                low = mid + 1
            
            if (arr[mid] == target):
                index = mid
        return index
    
    def find_right_index(self, arr, target):
        index = -1
        low = 0
        high = len(arr) -1
        while (low <= high):
            mid = int(low + (high - low) / 2)
            if (arr[mid] <= target):
                low = mid + 1
            else:
                high = mid - 1
            
            if (arr[mid] == target):
                index = mid
        return index

File: problems/034-first-last-position-in-sorted-array/main.cpp
class Solution {
public:
  int findLeftIndex(vector<int>& v, int target) {
    int index = -1;
    int low = 0;
    int high = v.size() - 1;
    while (low <= high) {
      int mid = low + (high - low) / 2;
      if (v[mid] >= target) {
        high = mid - 1;
      } else {
        low = mid + 1;
      }
      if (v[mid] == target) {
        index = mid;
      }
    }
    return index;
  }

  int findRightIndex(vector<int>&v, int target) {
    int index = -1;
    int low = 0;
    int high = v.size() - 1;
    while (low <= high) {
      int mid = low + (high - low) / 2;
      if (v[mid] <= target) {
        low = mid + 1;
      } else {
        high = mid - 1;
      }
      if (v[mid] == target) {
        index = mid;
      }
    }
    return index;
  }
  vector<int> searchRange(vector<int>& nums, int target) {
      vector<int> result(2, -1);
      result[0] = findLeftIndex(nums, target);
      result[1] = findRightIndex(nums, target);
      return result;
  }
};
